### ISSUE TITLE
{version} wildcard support

### ISSUE THREAD

**Issue by SasSam** [Stolz/Assets#61]
Hi!

Is there any chance to implement a {version} wildcard support when we define the bundle (collection)? Like in C# MVC4...

http://www.asp.net/mvc/overview/performance/bundling-and-minification
http://johnnewcombe.net/blog/mvc-4-part-4
http://stackoverflow.com/questions/12029161/version-wildcard-in-mvc4-bundle

Thanks!

**Comment by Stolz** [comment]
Wow, that is a lot of reading and I'm not a C# lover. Could you please summarize what it does? or give an example on how would you like to add your assets with wildcards and what will be the the expected behavior?

**Comment by SasSam** [comment]
In nutshell if you define a collection, you are able to use wild card character for filename definition like  the following 2 examples:

1) `{version}` wild card

    bundles.Add(new ScriptBundle("~/bundles/jquery").Include("~/Scripts/jquery-{version}.js"));

The preceding code creates a new JavaScript bundle named `~/bundles/jquery` that includes all the appropriate (that is debug or minified but not .vsdoc) files in the Scripts folder that match the wild card string `"~/Scripts/jquery-{version}.js"`. For ASP.NET MVC 4, this means with a  debug configuration, the file `jquery-1.7.1.js` will be added to the bundle. In a release configuration, `jquery-1.7.1.min.js` will be added. The bundling framework follows several common conventions such as:
* Selecting “.min” file for release when “FileX.min.js” and “FileX.js” exist.
* Selecting the non “.min” version for debug.
* Ignoring “-vsdoc” files (such as jquery-1.7.1-vsdoc.js), which are used only by IntelliSense.

The `{version}` wild card matching shown above is used to automatically create a jQuery bundle with the appropriate  version of jQuery in your Scripts folder.  In this example, using a wild card provides the following benefits:
* Allows you to use NuGet to update to a newer jQuery version without changing the preceding bundling code or jQuery references in your view pages.
* Automatically selects the full version for debug configurations and the ".min" version for release builds.

2) `*` wild card

`Include("~/Scripts/Common/*.js")` - AddAltToImg.js, ToggleDiv.js, ToggleImg.js
`Include("~/Scripts/Common/T*.js")` - Invalid pattern exception. The wildcard character is only allowed on the prefix or suffix.
`Include("~/Scripts/Common/*og.*")` - Invalid pattern exception. Only one wildcard character is allowed.
`Include("~/Scripts/Common/T*")` - ToggleDiv.js, ToggleImg.js
`Include("~/Scripts/Common/*")` - Invalid pattern exception. A pure wildcard segment is not valid.

The `Include` method can accept one "*" wildcard character as a  prefix or suffix to in the last path segment.

**Comment by Stolz** [comment]
The `*` wildcard is already implemented, take a look at the function [adDdir](https://github.com/Stolz/Assets/blob/master/API.md#adddir) at the docs. The `{version}` wildcard for sure wont get implemented because being this a framework agnostic package I don't think it should be aware of which current environment is it running.

**Comment by SasSam** [comment]
I've tried the `addDir()` but for some reason it drop back wrong path for the assets. This is my asset definition:

    Assets::addDir('js', '/(script-)(\d+)(\.min\.js)/');

And the `js_dir` is default `'js'`.

It find the file in the dir but when I check the generated and inserted link URL in HTML code, it's wrong:

    http://www.dev.local/js//js/script-123456.min.js 

If I use the normal `Assets::addJs('script-123456.min.js')` then it's working.

I've tried the following variations as well (all get back with same wrong result):

    Assets::addDir('js/', '/(script-)(\d+)(\.min\.js)/');
    Assets::addDir('', '/(script-)(\d+)(\.min\.js)/');
    Assets::addDir('./', '/(script-)(\d+)(\.min\.js)/');

What am I doing wrong?

I'm using the v0.0.9 because of Laravel 4.2.

Thanks!
